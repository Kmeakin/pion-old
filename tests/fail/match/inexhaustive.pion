//~ run = "${PION} elab ${FILE}"

fun (x : Bool) => match x {
    false => 1,
}
