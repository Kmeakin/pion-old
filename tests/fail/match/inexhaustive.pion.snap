exit status: 0
stdout:
fun (x : Bool) => match x {false => 1, _ => #error} : Bool -> Int

stderr:
error: inexhaustive match expression
  ┌─ tests/fail/match/inexhaustive.pion:3:25
  │  
3 │   fun (x : Bool) => match x {
  │                           ^ patterns not covered
  │ ╭───────────────────'
4 │ │     false => 1,
5 │ │ }
  │ ╰─' in match expression

